src/js/auth.js

```
@@ -0,0 +1,36 @@
+import { nanoid } from 'nanoid';
+
+const auth = () => {
+  const getSession = () => {
+    const userId = sessionStorage.getItem('userId');
+    const email = sessionStorage.getItem('email');
+    const date = new Date().toISOString();
+    sessionStorage.setItem('createdAt', date);
+
+    if (userId) {
+      return { userId, email, createdAt: date };
+    }
+    return null;
+  };
+
+  const signOut = () => {
+    sessionStorage.removeItem('userId');
+    sessionStorage.removeItem('email');
+    sessionStorage.removeItem('createdAt');
+  };
+
+  const signIn = (email) => {
+    let userId = sessionStorage.getItem('userId');
+
+    if (!userId) {
+      userId = nanoid();
+      sessionStorage.setItem('userId', userId);
+    }
+
+    sessionStorage.setItem('email', email);
+  };
+
+  return { getSession, signIn, signOut };
```

`signIn` should return the session object when the `signIn` is successful, else, it should return an error object, with the error message.

Please implement a `signUp` function that would create a new session with a new user if the user do not already exsit in the `sessionStorage` so that you can return that function. `signUp` should also return a session object.